src/Authorization.AspNetCore/DefaultClaimsPrincipalAccessor.cs

```diff
@@ -0,0 +1,33 @@
+using System.Security.Claims;
+using GraphQL.Validation;
+using Microsoft.AspNetCore.Http;
+
+namespace GraphQL.Server.Authorization.AspNetCore
+{
+    /// <summary>
+    /// The default claims principal accessor.
+    /// </summary>
+    public class DefaultClaimsPrincipalAccessor: IClaimsPrincipalAccessor
+    {
+        private readonly IHttpContextAccessor _contextAccessor;
+
+        /// <summary>
+        /// Creates an instance of <see cref="DefaultClaimsPrincipalAccessor"/>.
+        /// </summary>
+        /// <param name="contextAccessor">ASP.NET Core <see cref="IHttpContextAccessor"/> to take claims principal (<see cref="HttpContext.User"/>) from.</param>
+        public DefaultClaimsPrincipalAccessor(IHttpContextAccessor contextAccessor)
+        {
+            _contextAccessor = contextAccessor;
+        }
+
+        /// <summary>
+        /// Returns the <see cref="HttpContext.User"/>.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public ClaimsPrincipal GetClaimsPrincipal(ValidationContext context)
+        {
+            return _contextAccessor.HttpContext.User;
```

Comment:
```suggestion
            return _contextAccessor.HttpContext?.User;
```

It will be better if the call takes place outside the HTTP context. At the very least, we will not get NullReferenceException.